### PR TITLE
Validation of transaction instructions on world state view clone before commit to store.

### PR DESCRIPTION
**Pull request by humb1t**

### CODE REVIEW

**Comment by humb1t** [comment]
Depends on #467

File: iroha/src/tx.rs [hunk 1]
```diff
@@ -111,12 +116,18 @@ impl Transaction {
...
         if let Transaction::Signed {
             request,
             signatures,
         } = self
         {
+            let mut world_state_view = world_state_view.clone();
+            for instruction in &request.instructions {
+                if let Err(e) = instruction.invoke(&mut world_state_view) {
+                    return Err(e);
+                }
```
Comment (eadventurous) [inline iroha/src/tx.rs:129]: Can't we use `instruction.invoke(&mut world_state_view)?;` instead of `if let`?

File: iroha/src/wsv.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 use crate::prelude::*;
 use std::collections::HashMap;
 
-#[derive(Debug, Default)]
+#[derive(Debug, Clone, Default)]
```
Comment (eadventurous) [inline iroha/src/wsv.rs:4]: I haven't found where you `clone` WSV we usually clone `Arc` that references WSV, so do we really need it to be `Clone`?

File: iroha/src/tx.rs (hunk 1 above)
Comment (humb1t) [reply iroha/src/tx.rs:129]: You are right!

File: iroha/src/wsv.rs (hunk 2 above)
Comment (humb1t) [reply iroha/src/wsv.rs:4]: We clone the whole WSV in `validate` as WA for now

**Comment by humb1t** [comment]
Smthg wrong with CI

**Pull request closed without merging**